TSOClient/tso.client/UI/Panels/UITerrainHighlight.cs

```diff
@@ -24,9 +24,25 @@ public class UITerrainHighlight
             spriteBatch.Draw(Fill, new Rectangle((int)Start.X, (int)Start.Y, (int)length, lineWidth), null, tint, direction, new Vector2(0, 0.5f), SpriteEffects.None, 0); //
         }
 
-        public static void DrawArrow(UISpriteBatch batch, Terrain terrain, Vector2 from, int location)
+        public static Vector2 GetEndpointFromLotId(Terrain terrain, Vector2 from, int location) // allows us to find the end point of an arrow 
```

Comment:
If you're splitting this out, you should make the original function use this method rather than duplicating code.